### ISSUE TITLE
CB labels do not get cached correctly

### ISSUE THREAD

**Issue by lokitoth** [VowpalWabbit/vowpal_wabbit#3139]
CB labels do not get cached correctly: Their weight does not get cached.

data.txt:
```vw-text
shared | a:1 b:0.5
0:0.1:0.75 | a:0.5 b:1 c:2
| a:1 c:3

shared | s_1 s_2
0:1.0:0.5 | a:1 b:1 c:1
| a:0.5 b:2 c:1
```

When running VW without the cache (or the when generating the cache) the output is:
```
finished run
number of examples = 2
weighted example sum = 2.000000
weighted label sum = 0.000000
average loss = 1.066667
total feature number = 21
```
Command line = `--cb_adf -c -k -d data.txt`

When running VW the second time (with cache):
```
finished run
number of examples = 2
weighted example sum = 0.000000
weighted label sum = 0.000000
average loss = n.a.
total feature number = 21
```
Command line = `--cb_adf -c -d data.txt`

**Comment by lokitoth** [comment]
This has probably been going on a while, but since we were not using the weight field (until recently with #3055), it was invisible.

**Comment by lalo** [comment]
https://github.com/VowpalWabbit/vowpal_wabbit/blob/fb1a5a817634f445e3d73c0a5550d5449060129a/vowpalwabbit/cache.cc#L67
